Tests/SwiftDocCTests/Semantics/MetadataAlternateRepresentationTests.swift

```diff
@@ -0,0 +1,67 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2024 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Foundation
+import XCTest
+import Markdown
+
+@testable import SwiftDocC
+
+class MetadataAlternateRepresentationTests: XCTestCase {
+    func testValidLocalLink() throws {
+        for link in ["``MyClass/property``", "MyClass/property"] {
+            let (problems, metadata) = try parseDirective(Metadata.self) {
+                """
+                @Metadata {
+                    @AlternateRepresentation(\(link))
+                }
+                """
+            }
+            
+            XCTAssertTrue(problems.isEmpty, "Found problems: \(problems.joined(separator: ", "))")
+            XCTAssertEqual(metadata?.alternateRepresentations.count, 1)
+            
+            let alternateRepresentation = try XCTUnwrap(metadata?.alternateRepresentations.first)
+            XCTAssertEqual(alternateRepresentation.reference.url, URL(string: "MyClass/property"))
+        }
+    }
+        
+    func testValidExternalLinkReference() throws {
+        let (problems, metadata) = try parseDirective(Metadata.self) {
+            """
+            @Metadata {
+                @AlternateRepresentation("doc://com.example/documentation/MyClass/property")
+            }
+            """
+        }
+        
+        XCTAssertTrue(problems.isEmpty, "Found problems: \(problems.joined(separator: ", "))")
+        XCTAssertEqual(metadata?.alternateRepresentations.count, 1)
+        
+        let alternateRepresentation = try XCTUnwrap(metadata?.alternateRepresentations.first)
+        XCTAssertEqual(alternateRepresentation.reference.url, URL(string: "doc://com.example/documentation/MyClass/property"))
+    }
+
+    func testInvalidTopicReference() throws {
+        let (problems, _) = try parseDirective(Metadata.self) {
+            """
+            @Metadata {
+                @AlternateRepresentation("doc://")
+            }
+            """
+        }
+        
+        XCTAssertEqual(problems.count, 2, "Found problems: \(problems.joined(separator: ", "))")
+        XCTAssertEqual(problems, [
+            "1: note – org.swift.docc.Metadata.NoConfiguration",
+            "2: warning – org.swift.docc.HasArgument.unlabeled.ConversionFailed"
```

Comment:
The summary looks like the other ones for when the directive argument cannot be converted to a specific type:
```
Cannot convert 'doc://' to type 'TopicReference'"	
```

I wouldn't expect this warning to be hit a lot. The only cases where this will be hit is if it's a `doc://` link which is completely incorrectly formatted, like the example above. This won't be hit if the link cannot be resolved, that will be a different diagnostic. Most strings are recognised as valid topic references.

Reply:
Does it handle special characters or just what's allowed in a URL's path? For example; does `☜⃩/(_:_:)` convert to a TopicReference successfully?

Reply:
It's a syntactically valid Swift operator name, so it's good that it converts to a TopicReference